packages/models/src/lib/table.ts

```diff
@@ -1,33 +1,64 @@
 import { z } from 'zod';
 import { primitiveValueSchema } from './implementation/schemas';
 
-export const tableHeadingSchema = z.object(
+export const tableAlignmentSchema = z.enum(['left', 'center', 'right'], {
+  description: 'Cell alignment',
+});
+export type TableAlignment = z.infer<typeof tableAlignmentSchema>;
+
+export const tableColumnPrimitiveSchema = tableAlignmentSchema;
+export type TableColumnPrimitive = z.infer<typeof tableColumnPrimitiveSchema>;
+
+export const tableColumnObjectSchema = z.object({
+  key: z.string(),
+  label: z.string().optional(),
+  align: tableAlignmentSchema.optional(),
+});
+export type TableColumnObject = z.infer<typeof tableColumnObjectSchema>;
+
+export const tableRowObjectSchema = z.record(primitiveValueSchema, {
+  description: 'Object row',
+});
+export type TableRowObject = z.infer<typeof tableRowObjectSchema>;
+
+export const tableRowPrimitiveSchema = z.array(primitiveValueSchema, {
+  description: 'Primitive row',
+});
+export type TableRowPrimitive = z.infer<typeof tableRowPrimitiveSchema>;
+
+const tablePrimitiveSchema = z.object(
   {
-    key: z.string(),
-    label: z.string().optional(),
+    columns: z.array(tableAlignmentSchema).optional(),
+    rows: z.array(tableRowPrimitiveSchema),
   },
-  { description: 'Table heading' },
+  { description: 'Table with primitive rows and optional alignment columns' },
 );
-export type TableHeading = z.infer<typeof tableHeadingSchema>;
 
-export const tableAlignmentSchema = z.enum(['l', 'c', 'r'], {
-  description: 'Cell alignment',
-});
-export type TableAlignment = z.infer<typeof tableAlignmentSchema>;
+const tableObjectColPSchema = z.object(
+  {
+    columns: z.array(tableAlignmentSchema).optional(),
+    rows: z.array(tableRowObjectSchema),
+  },
+  {
+    description:
+      'Table with object rows and optional alignment or object columns',
+  },
+);
 
-export const tableRowSchema = z.union([
-  z.array(primitiveValueSchema),
-  z.record(primitiveValueSchema),
-]);
-export type TableRow = z.infer<typeof tableRowSchema>;
+const tableObjectColOSchema = z.object(
+  {
+    columns: z.array(tableColumnObjectSchema).optional(),
+    rows: z.array(tableRowObjectSchema),
+  },
+  {
+    description:
+      'Table with object rows and optional alignment or object columns',
+  },
+);
 
 export const tableSchema = (description = 'Table information') =>
-  z.object(
-    {
-      headings: z.array(tableHeadingSchema).optional(),
-      alignment: z.array(tableAlignmentSchema).optional(),
-      rows: z.array(tableRowSchema),
-    },
+  z.union(
+    [tablePrimitiveSchema, tableObjectColPSchema, tableObjectColOSchema],
     { description },
   );
```

Comment:
Would be simpler to merge `tableObjectColPSchema` and `tableObjectColOSchema` into one schema:

```ts
const tableObjectSchema = z.object(
  {
    columns: z.union([
      z.array(tableColumnObjectSchema),
      z.array(tableColumnPrimitiveSchema),
    ]).optional(),
    rows: z.array(tableRowObjectSchema),
  },
  {
    description:
      'Table with object rows and optional alignment or object columns',
  },
);
```

Reply:
merged them